### PR TITLE
Consolepasswordendpoint

### PR DESCRIPTION
**Pull request by majst01**

### CODE REVIEW

File: machine.go [hunk 1]
```diff
@@ -249,6 +249,16 @@ func (d *Driver) MachineGet(id string) (*MachineGetResponse, error) {
 	return response, nil
 }
 
+// MachineConsolePassword returns the consolepassword of the machine
+func (d *Driver) MachineConsolePassword(id, reason string) (*models.V1MachineConsolePasswordResponse, error) {
+	cp := machine.NewGetMachineConsolePasswordParams()
+	cp.Body.ID = &id
```
Comment (Gerrit91) [inline machine.go:255]: I am retrieving a nil pointer here:

```
❯ m machine consolepassword --reason "I want to test it" e0ab02d2-27cd-5a5e-8efc-080ba80cf258
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x979cfe]

goroutine 1 [running]:
github.com/metal-stack/metal-go.(*Driver).MachineConsolePassword(0xc000198460, 0x7ffd6795f1cc, 0x24, 0x7ffd6795f1ba, 0x11, 0x0, 0x0, 0xc00076ad00)
        /home/gerrit/go/pkg/mod/github.com/metal-stack/metal-go@v0.14.1-0.20210525094923-7b9e94ac3ed9/machine.go:255 +0xbe
github.com/metal-stack/metalctl/cmd.machineConsolePassword(0xc000198460, 0xc0004ce690, 0x1, 0x3, 0x0, 0x0)
        /home/gerrit/git/github.com/metal-stack/metalctl/cmd/machine.go:680 +0xc5
...
```

Comment (majst01) [reply machine.go]: fixed

File: machine.go [hunk 2]
```diff
@@ -249,6 +249,18 @@ func (d *Driver) MachineGet(id string) (*MachineGetResponse, error) {
...
+// MachineConsolePassword returns the consolepassword of the machine
+func (d *Driver) MachineConsolePassword(id, reason string) (*models.V1MachineConsolePasswordResponse, error) {
+	cp := machine.NewGetMachineConsolePasswordParams()
+	cp.Body = &models.V1MachineConsolePasswordRequest{
+		ID:     &id,
+		Reason: &reason,
+	}
+
+	resp, err := d.machine.GetMachineConsolePassword(cp, nil)
+	return resp.Payload, err
```
Comment (Gerrit91) [inline machine.go:261]: Needs `if err != nil` to prevent nil pointer. We're getting close. :)

Comment (majst01) [reply machine.go]: done

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**